src/admin.rs

```diff
@@ -106,312 +101,549 @@ impl AdminScClient {
+    }
 
-        client_w.create_topic(topic,replica.0,false).await
+    #[node_bindgen]
+    async fn list_spu(&self) -> Result<ArrayBuffer, FluvioError> {
+        if let Some(client) = self.inner.clone() {
+            let mut client = client.write().await.admin().await;
+            let spus = client.list::<SpuSpec, _>(vec![]).await?;
+            let json_slice = serde_json::to_vec(&spus).map_err(|err| {
+                FluvioError::Other(format!("serialization error: {}", err.to_string()))
+            })?;
+            Ok(ArrayBuffer::new(json_slice))
+        } else {
+            Err(FluvioError::Other("failed to lock client".to_owned()))
+        }
+    }
 
+    #[node_bindgen]
+    async fn find_partition(&self, topic: String) -> Result<PartitionMetadataWrapper, FluvioError> {
+        if let Some(client) = self.inner.clone() {
+            let mut client = client.write().await.admin().await;
+            let partitions = client.list::<PartitionSpec, _>(vec![]).await?;
+
+            if let Some(partition) = partitions.into_iter().find(|partition| {
+                let replica: ReplicaKey = partition
+                    .name
+                    .clone()
+                    .try_into()
+                    .expect("canot parse partition");
+
+                return replica.topic == topic && replica.partition == 0;
+            }) {
+                debug!("Found Partition: {:?}", partition);
+                Ok(PartitionMetadataWrapper(partition))
+            } else {
+                Err(FluvioError::Other("failed to find partition".to_owned()))
+            }
+        } else {
+            Err(FluvioError::Other("failed to lock client".to_owned()))
+        }
     }
 
     #[node_bindgen]
-    async fn delete_topic(&self,topic: String) -> Result<String,ClientError> {
+    async fn list_partitions(&self) -> Result<ArrayBuffer, FluvioError> {
+        if let Some(client) = self.inner.clone() {
+            let mut client = client.write().await.admin().await;
+            let partitions = client.list::<PartitionSpec, _>(vec![]).await?;
+            let json_slice = serde_json::to_vec(&partitions).map_err(|err| {
+                FluvioError::Other(format!("serialization error: {}", err.to_string()))
+            })?;
+            Ok(ArrayBuffer::new(json_slice))
+        } else {
+            Err(FluvioError::Other("failed to lock client".to_owned()))
+        }
+    }
 
-        let client = self.inner.as_ref().unwrap().clone();
-        let mut client_w = client.write().await;
+    #[node_bindgen]
+    async fn create_custom_spu(
+        &self,
+        name: String,
+        spec: CustomSpuSpecWrapper,
+    ) -> Result<(), FluvioError> {
+        if let Some(client) = self.inner.clone() {
+            let mut client = client.write().await.admin().await;
+            client.create(name, false, spec.0).await?;
+            Ok(())
+        } else {
+            Err(FluvioError::Other("failed to lock client".to_owned()))
```

Comment:
Uninitialized client, not failed to lock

Reply:
commit 0cc483b fixes;